plugins/inventory/scaleway.py

```diff
@@ -278,13 +288,30 @@ def do_zone_inventory(self, zone, token, tags, hostname_preferences):
                 # Composed variables
                 self._set_composite_vars(self.get_option('variables'), host_infos, hostname, strict=False)
 
+    def get_oauth_token(self):
+        oauth_token = self.get_option('oauth_token')
+        scw_config = os.path.join(os.path.expanduser('~'), '.config', 'scw', 'config.yaml')
+
+        if not oauth_token and os.path.exists(scw_config):
+            with open(scw_config) as fh:
+                values = yaml.safe_load(fh)
+                active_profile = values.get('active_profile', 'default')
+                if active_profile == 'default':
+                    oauth_token = values.get('secret_key')
+                else:
+                    oauth_token = values['profiles'][active_profile].get('secret_key')
+
+        return oauth_token
```

Comment:
In case neither the config file is around (or doesn't contain the token) nor the token has been provided by the user, the plugin needs to fail appropriately.